node/src/service.rs

```diff
@@ -340,30 +377,69 @@ async fn start_node_impl(
             overseer_handle.clone(),
             collator_key.clone(),
             parachain_consensus.clone(),
-        );
+        ); */
+
+        // TODO: refactor and build something similar to params_generator inside node builder.
+        // Params for collate_on_tanssi closure
+        let spawn_handle = node_builder.task_manager.spawn_handle().clone();
+        let spawn_essential = node_builder.task_manager.spawn_essential_handle().clone();
+        let keystore_clone = node_builder.keystore_container.keystore().clone();
+        let telemetry_handle = node_builder.telemetry.as_ref().map(|t| t.handle()).clone();
+        let block_import_clone = block_import.clone();
+        let client_clone = node_builder.client.clone();
+        let prometheus_registry = node_builder.prometheus_registry.clone();
+        let relay_interface = relay_chain_interface.clone();
+        let tx_pool = node_builder.transaction_pool.clone();
+        let sync_service = node_builder.network.sync_service.clone();
+        let collator_key_clone = collator_key.clone();
+        let para_id_clone = para_id;
+        let overseer = overseer_handle.clone();
+        let announce_block_clone = announce_block.clone();
 
         // TODO: change for async backing
         collate_on_tanssi = Some(move || async move {
-            #[allow(deprecated)]
-            cumulus_client_collator::start_collator(params_generator()).await;
-        });
-
-        node_builder
-            .start_collator(
-                para_id,
-                relay_chain_interface.clone(),
+            //#[allow(deprecated)]
+            //cumulus_client_collator::start_collator(params_generator()).await;
+            start_consensus_orchestrator(
+                client_clone,
+                block_import_clone,
+                prometheus_registry,
+                telemetry_handle.clone(),
+                spawn_handle,
+                spawn_essential,
+                relay_interface,
+                tx_pool,
+                sync_service,
+                keystore_clone.clone(),
+                force_authoring,
                 relay_chain_slot_duration,
-                parachain_consensus,
-                collator_key,
+                para_id_clone,
+                collator_key_clone,
+                overseer,
+                announce_block_clone,
             )
-            .await?
-    } else {
-        node_builder.start_full_node(
-            para_id,
+            .expect("Start consensus should succeed");
+        });
+
+        start_consensus_orchestrator(
```

Comment:
If collate_on_tanssi will do exactly the same as this, you can call collate_on_tanssi here, this way avoiding all the clones

```suggestion
        let call = collate_on_tanssi.clone().unwrap();
        call().await;
```